File: Object-Oriented Programming/Simulare/Teste.cpp
#include "Domain.h"
#include "Repository.h"
#include "Service.h"
#include <assert.h>

void teste_domain() {

	Carte c("a", "a", "a", 1);

	assert(c.get_autor() == "a");
	assert(c.get_culoare() == "a");
	assert(c.get_grosime() == 1);
	assert(c.get_titlu() == "a");
}

void teste_repo() {

	Repository_carti repo("teste.txt");

	vector<Carte> l = repo.get_all();

	assert(l.size() == 3);

	Carte c = repo.get_item_by_title("a");

	assert(c.get_autor() == "a");

	Repository_carti repo_err("");
}

void teste_service() {

	Repository_carti repo("teste.txt");

	Service_carti srv(repo);

	vector<Carte> l = srv.get_all();

	assert(l.size() == 3);

	Carte c = srv.get_item_by_title("a");

	assert(c.get_autor() == "a");

	vector<Carte> rez_s1 = srv.sortare_dupa_grosime();

	assert(rez_s1[0].get_grosime() == 1);
	assert(rez_s1[1].get_grosime() == 2);
	assert(rez_s1[2].get_grosime() == 3);

	vector<Carte> rez_s2 = srv.sortare_dupa_titlu();

	assert(rez_s2[0].get_titlu() == "a");
	assert(rez_s2[1].get_titlu() == "b");
	assert(rez_s2[2].get_titlu() == "c");
}

void ruleaza_teste() {

	teste_domain();
	teste_repo();
	teste_service();
}

File: FundamentalsOfProgramming/Fundamentele Programarii Lucru Suplimentar/FP biblioteca/teste.py
from domain import Carte,ValidatorCarteException,ValdiatorCarte
from repository_carti import Repository,RepositoryException
from service import Service,ServiceException
import unittest

class Teste_domain(unittest.TestCase):

    def setUp(self):

        self.c = Carte(1,"a","a",1)
    
    def tearDown(self):

        self.c = Carte(1,"a","a",1)
    
    def testGetteri(self):

        self.assertEqual(self.c.getId(),1)
        self.assertEqual(self.c.getTitlu(),"a")
        self.assertEqual(self.c.getAutor(),"a")
        self.assertEqual(self.c.getAn_aparitie(),1)
    
    def testSetter(self):

        self.c.setAutor("b")
        self.assertEqual(self.c.getAutor(),"b")
    
    def testValdiator(self):

        v = ValdiatorCarte()
        
        self.assertRaises(ValidatorCarteException,v.validate,Carte(-1,"a","a",1))
        with self.assertRaises(ValidatorCarteException):
            v.validate(Carte(1,"","a",1))
            v.validate(Carte(1,"a","",1))
            v.validate(Carte(1,"a","a",-1))
        
        self.assertIsNone(v.validate(Carte(1,"a","a",1)))

class Teste_repo(unittest.TestCase):

    def setUp(self):

        try:
            f = open("biblioteca_test.txt","w")
        except IOError:
            return
        f.write("1; Harap Alb; dadaaa; 1974\n2; Carte; Autor; 2012\n3; Mov; Hautor; 2013\n10; da; dadaaa; 10\n")
        f.close()
        
        self.repo = Repository("biblioteca_test.txt")
    
    def tearDown(self):

        pass
    
    def test_read_from_file(self):

        l = self.repo.getAll()
        self.assertTrue(len(l) == 4)
        try:
            f = open("biblioteca_test.txt","w")
        except IOError:
            return
        f.write("1; Harap Alb; dadaaa; 1974\n3; Mov; Hautor; 2013\n10; da; dadaaa; 10\n")
        f.close()
        repo = Repository("biblioteca_test.txt")
        l = repo.getAll()
        self.assertTrue((len(l) == 3))
    
    def test_write_to_file(self):

        l = self.repo.getAll()
        self.assertTrue(len(l) == 4)
        self.repo.adaugare_carte(Carte(100,"a","a",100))

        repo = Repository("biblioteca_test.txt")
        l = repo.getAll()
        self.assertTrue((len(l) == 5))